src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -224,6 +224,11 @@ public override void VisitLocalFunctionStatement(LocalFunctionStatementSyntax no
                 var oldMethod = _containingMemberOrLambda;
                 _containingMemberOrLambda = match;
 
+                foreach (var parameter in node.ParameterList.Parameters)
+                {
+                    Visit(parameter.Default);
```

Comment:
You make it sound like the binders that are created in LocalBinderFactory.cs are never used to bind default parameters, at the same time creating the binders avoids a crash. So, to me it looks like, we are binding default parameters with binders that differ in flags, i.e. inconsistently.